nextcloud/datadog_checks/nextcloud/nextcloud.py

```diff
@@ -43,13 +41,10 @@ class NextcloudCheck(AgentCheck):
 
     def check(self, instance):
         url = instance['url']
-        username = instance['username']
-        password = instance['password']
-        auth = (username, password)
 
         try:
             self.log.debug("Checking against %s", url)
-            response = requests.get(url, auth=auth, headers=headers(self.agentConfig))
+            response = self.http.get(url, extra_headers=headers(self.agentConfig))
```

Comment:
why is this sending the agent config as a header?

Reply:
How is this going to work without the auth?

Reply:
That is what the requests wrapper does, it handles auth and proxies. You can read more about it in https://datadoghq.dev/integrations-core/base/http/